src/Features/Core/Portable/EditAndContinue/AbstractEditAndContinueAnalyzer.cs

```diff
@@ -3708,6 +3710,58 @@ private static bool TryMapParameter((SyntaxNode? Node, int Ordinal) parameterKey
             }
         }
 
+        protected virtual void ReportLambdaAttributeRudeEdits(SemanticModel oldModel,
+            SyntaxNode oldLambdaBody,
+            SemanticModel newModel,
+            SyntaxNode newLambdaBody,
+            List<RudeEditDiagnostic> diagnostics,
+            bool hadSignatureEdits,
+            CancellationToken cancellationToken)
+        {
+            var newLambda = GetLambda(newLambdaBody);
+            var oldLambda = GetLambda(oldLambdaBody);
+
+            Debug.Assert(IsNestedFunction(newLambda) == IsNestedFunction(oldLambda));
+
+            // queries are analyzed separately
+            if (!IsNestedFunction(newLambda))
+            {
+                return;
+            }
+
+            var oldLambdaSymbol = GetLambdaExpressionSymbol(oldModel, oldLambda, cancellationToken);
+            var newLambdaSymbol = GetLambdaExpressionSymbol(newModel, newLambda, cancellationToken);
+
+            var reportDiagnostic = false;
+            if (!oldLambdaSymbol.GetAttributes().SequenceEqual(newLambdaSymbol.GetAttributes()) ||
+                !oldLambdaSymbol.GetReturnTypeAttributes().SequenceEqual(newLambdaSymbol.GetReturnTypeAttributes()))
+            {
+                reportDiagnostic = true;
+            }
+
+            // If the old and new signatures are the same then we are safe to compare parameter attributes more easily
```

Comment:
>are safe to compare parameter attributes more easily [](start = 66, length = 52)

I'd say that we don't need to report rude edits for parameter attributes if the signatures don't match since we already reported rude edit for the signature mismatch

Reply:
Yes, that is what this is doing.. I'll reword the comment.